3rdparty/openssl/deprecations.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_OPENSSL_DEPRECATIONS_H
+#define _OE_OPENSSL_DEPRECATIONS_H
+
+#if !defined(OE_OPENSSL_SUPPRESS_DEPRECATIONS)
+
+#if defined(__cplusplus)
+#define OE_OPENSSL_EXTERN_C_BEGIN \
+    extern "C"                    \
+    {
+#define OE_OPENSSL_EXTERN_C_END }
+#else
+#define OE_OPENSSL_EXTERN_C_BEGIN
+#define OE_OPENSSL_EXTERN_C_END
+#endif
+
+#ifndef OE_DISABLE_OPENSSL_DEPRECATIONS
+#define OE_OPENSSL_DEPRECATED(MSG) __attribute__((deprecated(MSG)))
```

Comment:
From the discussions, it sounded like we wanted to not use the compiler deprecation mechanism and directly force compilation to break via macro redefinitions.

@anakrish?

Reply:
Changed to enforce errors instead of just warnings.